lib/uploadcare/utils/user_agent.rb

```diff
@@ -0,0 +1,35 @@
+module Uploadcare
+  # Determines User-Agent string either taking it from settings or building
+  # in accordance with common Uploadcare format
+  #
+  class UserAgent
+    # @param options [Hash]
+    # @option options [String] :user_agent (nil)
+    # @option options [String] :public_key (nil)
+    # @option options [String] :user_agent_extension (nil)
+    # @return [String]
+    #
+    def call(options)
+      return options[:user_agent].to_s if options[:user_agent]
+      user_agent_string(options)
+    end
+
+    private
+
+    def user_agent_string(options)
+      format(
+        '%<library_string>s/%<public_key>s (%<environment_string>s)',
```

Comment:
it seems that 'UploadcareRuby/' can be moved to this line

Reply:
It can, but I've tried to visually separate User-Agent string format from the actual data